gossip/src/contact_info.rs

```diff
@@ -1016,4 +1024,59 @@ mod tests {
             Err(Error::InvalidPort(0))
         );
     }
+
+    #[test]
+    fn test_check_duplicate() {
+        let mut rng = rand::thread_rng();
+        let mut node = ContactInfo::new(
+            Keypair::new().pubkey(),
+            rng.gen(), // wallclock
+            rng.gen(), // shred_version
+        );
+        // Same contact-info is not a duplicate instance.
+        {
+            let other = node.clone();
+            assert!(!node.check_duplicate(&other));
+            assert!(!other.check_duplicate(&node));
+        }
+        // Updated socket address is not a duplicate instance.
+        {
+            let mut other = node.clone();
+            other.set_gossip(new_rand_socket(&mut rng)).unwrap();
+            other.set_serve_repair(new_rand_socket(&mut rng)).unwrap();
```

Comment:
Unwrap here causes test flakiness since random sockets could be invalid

Reverted this PR in https://github.com/anza-xyz/agave/pull/2521